src/librbd/internal.cc

```diff
@@ -3374,7 +3378,8 @@ namespace librbd {
   }
 
   int aio_read(ImageCtx *ictx, const vector<pair<uint64_t,uint64_t> >& image_extents,
-	       char *buf, bufferlist *pbl, AioCompletion *c, int op_flags)
+	       char *buf, bufferlist *pbl, AioCompletion *c, int op_flags,
+	       int force_no_cache)
```

Comment:
Hmm ... there is a different ```cache_lock``` between the parent and child images, so you shouldn't get deadlocked so long as you aren't re-reading from the same image within the ```AioRead::should_complete``` / ```AbstractWrite::should_complate```